packages/core/src/components/autosuggest/autosuggest.tsx

```diff
@@ -195,13 +201,31 @@ export class Autosuggest {
       this.input.addEventListener("input", this.onInput);
       this.input.addEventListener("keydown", this.onKeyDown);
       this.input.addEventListener("focusin", this.onFocusIn);
+
+      window.addEventListener("resize", this.onWindowResize);
```

Comment:
Zie mijn vorige opmerking over `ResizeObserver`, dat is een betere kandidaat om te bepalen of een component moet re-renderen.

Reply:
Ja, dat heb ik geprobeerd, maar werkt niet in alle gevallen. Weet niet meer welke. Zie wel dat in `header.tsx` ook geen `ResizeObserver` wordt gebruikt:

```typescript
connectedCallback() {
    window.addEventListener("resize", this.onWindowResize);
  }

  disconnectedCallback() {
    window.removeEventListener("resize", this.onWindowResize);
  }
```

Reply:
Volgens mij maakt het niet uit wat er met de viewport gebeurt, maar met de breedte van het element en kan je met `ResizeObserver` het host element in de gaten houden.

* De viewport kan wijzigen zonder dat de breedte van het element wijzigt (impact: geen probleem, maar nodeloze herberekeningen). Bijvoorbeeld de viewer: Verklein de viewport maar het linker zoekpaneel blijft nog steeds 592px breed.
* De viewport wijzigt niet maar een scriptende routine maakt het element kleiner. (impact: Er wordt geen herberekening gedaan). Bijvoorbeeld de viewer: Verklein het zoekenpaneel met de kleiner/groter worden knop.

Kijk eens naar `ResizeObserver` zoals die in `image-overlay.tsx` wordt ingezet.

Reply:
Voor het nageslacht: Eric had een terecht punt dat `window.addEventListener("resize", this.onWindowResize);` óók nodig is: De lijst met autosuggesties moet binnen de viewport blijven, en dat kan niet met `ResizeObserver` worden gemeten.